File: week_2/박설화/크레인_인형뽑기_게임.js
function solution(board, moves) {
  let stack = [];
  let answer = 0;
  
  moves.forEach(el => {
      let i = el - 1;
      
      for(let m = 0; m < board.length; m++){
          let doll = board[m][i];
                          
          if(doll){
              board[m][i] = 0;
              if(stack.length && stack[stack.length - 1] === doll){
                  stack.pop();
                  answer += 2;
              } else {
                  stack.push(doll);
              }
              break;
          }
      }
  })

  return answer;
}

File: week_2/김문기/크레인 인형 뽑기 게임.py
def solution(board, moves):
    answer = 0
    basket = []
    n = len(board)
    
    for move in moves:
        col = move - 1  
        
        for row in range(n):
            if board[row][col] != 0:
                obj = board[row][col]
                board[row][col] = 0  
                
                if basket and basket[-1] == obj:
                    basket.pop()
                    answer += 2
                else:
                    basket.append(obj)
                break  
    
    return answer
